### PR TITLE
Added personal direct friend invite link endpoints

### PR DESCRIPTION
**Pull request by DeadLyBro**
Updated;

+ getFriendInvites()
+ createFriendInvite()
+ deleteFriendInvites()

getFriendInvites is returning your active Friend Invites.
createFriendInvite is giving Friend Invite for use 5 times. 
deleteFriendInvites is deleting your all Friend Invite.

You can use the "code". (Length 8 char.)
E.g: https://discord.gg/AaBbcCDd

I hope you guys understand what I said here.

### CODE REVIEW

**Comment by DeadLyBro** [comment]
I don't tried with "code" to `deleteFriendInvite()` , so i don't know if that is working for "code" delete. 
But I'm sure its deleting last one with that code.

**Comment by DeadLyBro** [comment]
Ups, I'm so sorry its deleting all invites lol. I check it now.

**Comment by rigwild** [comment]
Please also add types in https://github.com/rigwild/discord-self-bot-console/blob/main/types.d.ts

**Comment by DeadLyBro** [comment]
```ts
getFriendInvite(): Promise<any>
createFriendInvite(): Promise<any>
deleteFriendInvite(): Promise<any>
```

Is it ok like this? @rigwild

**Comment by rigwild** [comment]
Yes it's ok

Also add to delete 1 invite 

> Ups, I'm so sorry its deleting all invites lol. I check it now.

**Comment by DeadLyBro** [comment]
> Also add to delete 1 invite
> 
> > Ups, I'm so sorry its deleting all invites lol. I check it now.

Idk how to do that, can you try it?

**Comment by DeadLyBro** [comment]
Ok i found it.

**Comment by DeadLyBro** [comment]
I hope its work.

**Comment by DeadLyBro** [comment]
Oh no, I tried now... deleteFriendInvite(code) is deleting all invites again...

**Comment by DeadLyBro** [comment]
can you help me

**Comment by DeadLyBro** [comment]
We are getting a friend invite link with that code so you can use only like this; https://discord.gg/AaBbcCDd

**Comment by DeadLyBro** [comment]
image

**Comment by rigwild** [comment]
Ok but what are you achieving? Inviting what? To a guild? I'm confused as to what this does

**Comment by DeadLyBro** [comment]
No, with this invite link, someone can add a friend you without asking.

**Comment by DeadLyBro** [comment]
Deleting deleteFriendInvite because not working...

File: index.js [hunk 1]
```diff
@@ -147,6 +147,11 @@
     getDMs: () => apiCall(`/users/@me/channels`),
     getUser: userId => apiCall(`/users/${userId}`),
 
+    getFriendInvites: () => apiCall(`/users/@me/invites`),
+    createFriendInvite: () => apiCall(`/users/@me/invites`, null, 'POST'),
+    deleteFriendInvites: () => apiCall(`/users/@me/invites`, null, 'DELETE'),
+	  deleteFriendInvite: (code) => apiCall(`/users/@me/invites`, code, 'DELETE'),
+
```
Comment (rigwild) [inline index.js]:
```suggestion
    getFriendDirectInviteLinks: () => apiCall(`/users/@me/invites`),
    createFriendDirectInviteLink: () => apiCall(`/users/@me/invites`, null, 'POST'),
    deleteAllFriendDirectInviteLinks: () => apiCall(`/users/@me/invites`, null, 'DELETE'),
```

**Comment by DeadLyBro** [comment]
image
I found this on API... Gonna try it.

File: types.d.ts [hunk 2]
```diff
@@ -147,6 +147,11 @@ export type api = {
   getDMs(): Promise<any>
   getUser(userId: string): Promise<User>
 
+  getFriendInvites(): Promise<any>
+  createFriendInvite(): Promise<any>
+  deleteFriendInvites(): Promise<any>
+  deleteFriendInvite(code: string): Promise<any>
```
Comment (rigwild) [inline types.d.ts]:
```suggestion
  getFriendDirectInviteLinks(): Promise<any>
  createFriendDirectInviteLink(): Promise<any>
  deleteAllFriendDirectInviteLinks(): Promise<any>
```

**Comment by rigwild** [comment]
I looked on the web and found out that it's not a supported feature by discord and was abandonned quickly after they added it

But looks like it still works, though the invite expires after 7d or 5 uses maximum

For delete 1 invite I tried to call `DELETE /invites/code` which is the one they call to delete invite link in guilds, but does not work

Maybe they never implemented it

**Comment by DeadLyBro** [comment]
Yeah I think so. You can search on web; Discord Friend Invite. 
There is a video channel named No Text To Speech.

File: index.js [hunk 3]
```diff
@@ -147,6 +147,10 @@
     getDMs: () => apiCall(`/users/@me/channels`),
     getUser: userId => apiCall(`/users/${userId}`),
 
+    getFriendInvites: () => apiCall(`/users/@me/invites`),
+    createFriendInvite: () => apiCall(`/users/@me/invites`, null, 'POST'),
+    deleteFriendInvites: () => apiCall(`/users/@me/invites`, null, 'DELETE'),
```
Comment (rigwild) [inline index.js:152]:
```suggestion
    getDirectFriendInviteLinks: () => apiCall(`/users/@me/invites`),
    createDirectFriendInviteLink: () => apiCall(`/users/@me/invites`, null, 'POST'),
    deleteDirectFriendInviteLinks: () => apiCall(`/users/@me/invites`, null, 'DELETE'),
```

File: types.d.ts [hunk 4]
```diff
@@ -147,6 +147,10 @@ export type api = {
   getDMs(): Promise<any>
   getUser(userId: string): Promise<User>
 
+  getFriendInvites(): Promise<any>
+  createFriendInvite(): Promise<any>
+  deleteFriendInvites(): Promise<any>
```
Comment (rigwild) [inline types.d.ts:152]:
```suggestion
  getDirectFriendInviteLinks(): Promise<any>
  createDirectFriendInviteLink(): Promise<any>
  deleteDirectFriendInviteLinks(): Promise<any>
```

**Review by rigwild — CHANGES_REQUESTED** [review]

**Review by rigwild — APPROVED** [review]

**Comment by rigwild** [comment]
Thanks!

**Pull request merged**